docs/ingress-service-dns-with-coredns.md

```diff
@@ -0,0 +1,191 @@
+# Federation DNS for Ingress and Service
+
+This tutorial describes how to set up a federation cluster DNS with [ExternalDNS](https://github.com/kubernetes-incubator/external-dns/) based on [CoreDNS](https://github.com/coredns/coredns) in [minikube](https://github.com/kubernetes/minikube) clusters. It provides guidance for the following steps:
+
+- Install ExternalDNS with etcd enabled CoreDNS as a provider
+- Install [ingress controller](https://github.com/kubernetes/ingress-nginx) for your minikube clusters to enable Ingress resource
+- Install [metallb](https://github.com/google/metallb) for your minikube clusters to enable LoadBanlancer Service
+
+You can use either Loadbalancer Service or Ingress resource or both in your environment, this tutorial include guidance for both Loadbalancer Service and Ingress resource.
+For related conceptions of Muilti-cluster Ingress and Service, you can refer to [ingressdns-with-externaldns.md](https://github.com/kubernetes-sigs/federation-v2/blob/master/docs/ingressdns-with-externaldns.md) and [servicedns-with-externaldns.md](https://github.com/kubernetes-sigs/federation-v2/blob/master/docs/servicedns-with-externaldns.md).
+
+## Creating federation cluster
+
+Install Federation-v2 with minikube in [User Guide](https://github.com/kubernetes-sigs/federation-v2/blob/master/docs/userguide.md).
+
+## Installing ExternalDNS
+
+Install ExternalDNS with CoreDNS as backend in your host cluster. You can follow the [tutorial](https://github.com/kubernetes-incubator/external-dns/blob/master/docs/tutorials/coredns.md).
+
+To make it work for federation resources, you need to use below ExternalDNS deployment instead of the one in the tutorial.
+**Note**: You should replace value of `ETCD_URLS` with your own etcd client service IP address.
+
+```bash
+$ cat <<EOF | kubectl create -f -
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: external-dns
+  namespace: kube-system
+spec:
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app: external-dns
+  template:
+    metadata:
+      labels:
+        app: external-dns
+    spec:
+      containers:
+      - name: external-dns
+        image: registry.opensource.zalan.do/teapot/external-dns:latest
+        args:
+        - --source=crd
+        - --crd-source-apiversion=multiclusterdns.federation.k8s.io/v1alpha1
+        - --crd-source-kind=DNSEndpoint
+        - --registry=txt
+        - --provider=coredns
+        - --log-level=debug # debug only
+        env:
+        - name: ETCD_URLS
+          value: http://10.102.147.224:2379
+EOF
+```
+
+## Enable DNS for federation resources
+
+### Installing metellb for LoadBalancer Service
+
+Install metallb in each member cluster to make LoadBanlancer type Service work.
+
+```bash
+$ helm --kube-context cluster1 install --name metallb stable/metallb
+$ helm --kube-context cluster2 install --name metallb stable/metallb
+```
+
+Apply configmap to configure metallb in each cluster.
+
```

Comment:
I use different IP range configuration for this case for testing.
What is your proposal for such kind of IP range?

Reply:
My point is that no context is provided. Where do these addresses come from? Do they need to be routable between the clusters? What is `10.0.0.1`? etc.. I had time to go through the metallb [tutorial](https://metallb.universe.tf/tutorial/minikube/) and I think it might make more sense to reference that document instead of the current approach.

Reply:
I link metallb site for BGP on Minikube in this document for reference and keep existing example configuration to help user set up enviornment conveniently.